src/subdataframe/subdataframe.jl

```diff
@@ -62,18 +62,14 @@ immutable SubDataFrame{T <: AbstractVector{Int}} <: AbstractDataFrame
     end
 end
 
-function SubDataFrame{T <: AbstractVector{Int}}(parent::DataFrame, rows::T)
-    return SubDataFrame{T}(parent, rows)
+function SubDataFrame{S <: Integer}(parent::DataFrame, rows::AbstractVector{S})
+    return SubDataFrame{AbstractVector{Int}}(parent, convert(AbstractVector{Int}, rows))
```

Comment:
`{AbstractVector{Int}}` isn't correct here: it should be the actual concrete type to which `rows` was converted. So you need to store the result of the conversion in an object first, and call `typeof` on it.